src/smtp/client.rs

```diff
@@ -234,14 +234,37 @@ mod test {
     #[tokio::test]
     async fn transparency_procedure() {
         for (test, result) in [
+            // escapes required
             (
                 "A: b\r\n.\r\n".to_string(),
                 "A: b\r\n..\r\n\r\n.\r\n".to_string(),
             ),
+            (
+                "A: b\r.\r\n".to_string(),
+                "A: b\r..\r\n\r\n.\r\n".to_string(),
+            ),
+            (
+                "A: b\n.\r\n".to_string(),
+                "A: b\n..\r\n\r\n.\r\n".to_string(),
+            ),
+            (
+                "A: b\r\n.\r\nSome text".to_string(),
+                "A: b\r\n..\r\nSome text\r\n.\r\n".to_string(),
+            ),
+            (
+                "A: b\r.\r\nSome text".to_string(),
+                "A: b\r..\r\nSome text\r\n.\r\n".to_string(),
+            ),
+            (
+                "A: b\n.\r\nSome text".to_string(),
+                "A: b\n..\r\nSome text\r\n.\r\n".to_string(),
+            ),
             ("A: b\r\n.".to_string(), "A: b\r\n..\r\n.\r\n".to_string()),
+
+            // no escapes required
```

Comment:
As per RFC5321 section 4.5.2, every line starting with a `.` needs to be escaped:

```
      Before sending a line of mail text, the SMTP client checks the
      first character of the line.  If it is a period, one additional
      period is inserted at the beginning of the line.
```

If this is not done the receiving mail server will now be able to restore the original message.